core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -906,6 +904,62 @@ void rename(RelNode rel) {
     }
   }
 
+  /**
+   * Checks whether a relexp has made any subset cheaper, and if it so,
+   * propagate new cost to parent rel nodes.
+   *
+   * @param planner   Planner
+   * @param mq        Metadata query
+   * @param rel       Relational expression whose cost has improved
+   */
+  void propagateCostImprovements(VolcanoPlanner planner,
+      RelMetadataQuery mq, RelNode rel) {
```

Comment:
`mq` can be retrieved from `rel.getCluster().getMetadataQuery()`, so mq is not needed.

Reply:
good point, removed